src/dotnet/App.Maui/MainPage.xaml.cs

```diff
@@ -11,14 +11,15 @@ public partial class MainPage : ContentPage
 
     public BlazorWebView BlazorWebView
         => this._blazorWebView;
+    private ILogger Log { get; }
 
     public MainPage(ClientAppSettings appSettings, NavigationInterceptor navInterceptor)
     {
         _appSettings = appSettings;
         _navInterceptor = navInterceptor;
+        Log = AppServices.LogFor<MainPage>();
```

Comment:
I think better to inject logger in constructor as well as ClientAppSettings and NavigationInterceptor.